indico/modules/events/registration/client/js/form/fields/TimetableSessions.jsx

```diff
@@ -0,0 +1,309 @@
+  minimum: PropTypes.number.isRequired,
+};
+
+TimetableSessionsComponent.defaultProps = {
+  sessionData: {},
+  display: 'expanded',
+};
+
+function SessionBlockHeader({
+  value,
+  index,
+  data,
+  label,
+  isExpanded,
+  handleValueChange,
+  onChange,
+  onClick,
+}) {
+  const childrenIds = data.map(e => e.id);
+  const isAllChecked = childrenIds.every(id => value.includes(id));
+  const isAnyChecked = childrenIds.some(id => value.includes(id));
+  const selectedBlocks = value.filter(element => childrenIds.includes(element)).length;
+
+  function handleHeaderChange(e) {
+    e.stopPropagation();
+    const newValue =
+      isAllChecked || isAnyChecked
+        ? value.filter(element => !childrenIds.includes(element))
+        : [...value, ...childrenIds];
+    onChange(newValue);
+  }
+
+  return (
+    <>
+      <AccordionTitle active={isExpanded} index={index} onClick={onClick}>
+        <Icon name="dropdown" />
+        <Checkbox
+          checked={isAllChecked}
+          onChange={handleHeaderChange}
+          indeterminate={isAnyChecked && !isAllChecked}
+          label={label}
+        />
+        <span>
+          <PluralTranslate count={selectedBlocks}>
+            <Singular>
+              (<Param name="selectedBlocks" value={selectedBlocks} /> block selected)
+            </Singular>
+            <Plural>
+              (<Param name="selectedBlocks" value={selectedBlocks} /> blocks selected)
+            </Plural>
+          </PluralTranslate>
+        </span>
+      </AccordionTitle>
+      <AccordionContent active={isExpanded}>
+        {data.map(({fullTitle, startTime, id}) => (
+          <dd className="grouped-fields" key={id}>
+            <Checkbox
+              value={id}
+              onChange={handleValueChange}
+              checked={value.includes(id)}
+              label={`${fullTitle}-${startTime}`}
+            />
+          </dd>
+        ))}
+      </AccordionContent>
+    </>
+  );
+}
+
+SessionBlockHeader.propTypes = {
+  value: PropTypes.arrayOf(PropTypes.number).isRequired,
+  index: PropTypes.number.isRequired,
+  data: sessionDataSchema.isRequired,
+  label: PropTypes.string.isRequired,
+  isExpanded: PropTypes.bool.isRequired,
+  handleValueChange: PropTypes.func.isRequired,
+  onChange: PropTypes.func.isRequired,
+  onClick: PropTypes.func.isRequired,
+};
+
+export default function TimetableSessionsInput({
+  htmlId,
+  htmlName,
+  disabled,
+  isRequired,
+  display,
+  minimum,
+  maximum,
+}) {
+  const {eventId} = useSelector(getStaticData);
+  const {data: sessionData} = useIndicoAxios(
+    {url: timeTableURL({event_id: eventId})},
+    {camelize: true}
+  );
+  let validationMsg;
+
+  if (minimum > 0 && maximum > 0) {
+    validationMsg = PluralTranslate.string(
+      'Please select at least {minimum} and no more than {maximum} session block',
+      'Please select at least {minimum} and no more than {maximum} session blocks',
+      maximum,
+      {minimum, maximum}
+    );
```

Comment:
This message is not translatable because it depends on two arguments (`minimum`, `maximum`) and we're only using one to select the corresponding plural (`minimum`). Luckily, in this case I think we can just remove it because a user might select too few or too many sessions but not both.

Then, for the validation, you'd have something like this:
```js
if (minimum !== 0 && value.length < minimum) {
  return minMsg;
} else if (maximum !== 0 && value.length > maximum) {
  return maxMsg;
}
```